src/test/java/de/sldk/mc/metrics/WorldSizeTest.java

```diff
@@ -0,0 +1,111 @@
+package de.sldk.mc.metrics;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+import io.prometheus.client.CollectorRegistry;
+import java.io.*;
+import java.nio.file.*;
+import java.util.Random;
+import net.bytebuddy.utility.RandomString;
+import org.bukkit.World;
+import org.bukkit.plugin.Plugin;
+import org.junit.jupiter.api.*;
+
+public class WorldSizeTest {
+
+    private static final String METRIC_NAME = "mc_world_size";
+    private static final String[] METRIC_LABELS = { "world" };
+
+    private WorldSize worldSizeMetric;
+
+    private Path path;
+    private File file;
+    private World world;
+
+    @BeforeEach
+    public void beforeEach() throws IOException{
+        worldSizeMetric = new WorldSize(mock(Plugin.class));
+        worldSizeMetric.enable();
+    }
+
+    @AfterEach
+    public void afterEach() {
+        CollectorRegistry.defaultRegistry.clear();
+        try {
+            Files.deleteIfExists(path);
+        } catch (Throwable t) { }
+    }
+
+    @Test
+    public void doesNotThrowIfFileDoesNotExist() throws IOException {
+        givenWorldFileDoesNotExist();
+        assertDoesNotThrow(() -> worldSizeMetric.collect(world));
+    }
+
+    @Test
+    public void doesNotPopulateMetricIfFileDoesNotExist() throws IOException {
+        givenWorldFileDoesNotExist();
+        worldSizeMetric.collect(world);
+        assertMetricIsEmpty();
+    }
+
+    @Test
+    public void setsMetricWithCorrectNameAndLabel() throws IOException {
+        String worldName = new RandomString(10).nextString();
+        givenWorldFileExists(worldName);
+        worldSizeMetric.collect(world);
+        assertNotNull(getMetricValue(worldName));
+    }
+
+    @Test
+    public void setsCorrectWorldSizeValue() throws IOException {
+        String worldName = new RandomString(10).nextString();
+        int worldSize = new Random().ints(128, 1024).findFirst().getAsInt();
+        givenWorldFileExists(worldName, worldSize);
+        worldSizeMetric.collect(world);
+        Double value = getMetricValue(worldName);
+        assertEquals(worldSize, value.longValue());
+    }
+
+    private Double getMetricValue(String worldName) {
+        return CollectorRegistry.defaultRegistry.getSampleValue(
+            METRIC_NAME,
+            METRIC_LABELS,
+            new String[] { worldName }
+        );
+    }
+
+    private void assertMetricIsEmpty() {
+        assertNull(CollectorRegistry.defaultRegistry.getSampleValue(METRIC_NAME));
+    }
+
+    private void givenWorldFileDoesNotExist() throws IOException {
```

Comment:
The `IOException` is never thrown here.

```suggestion
    private void givenWorldFileDoesNotExist() {
```